githubutil/githubutil.go

```diff
@@ -35,11 +42,115 @@ func NewClient(ctx context.Context, pat string) (*github.Client, error) {
 	return github.NewClient(tokenClient), nil
 }
 
+// NewInstallationClient creates a GitHub client using the given GitHub Client ID, installation ID, and private key.
+func NewInstallationClient(ctx context.Context, clientID string, installationID int64, privateKey string) (*github.Client, error) {
+	if clientID == "" {
+		return nil, errors.New("no GitHub App Client ID specified")
+	}
+	if installationID == 0 {
+		return nil, errors.New("no GitHub App Installation ID specified")
+	}
+	if privateKey == "" {
+		return nil, errors.New("no GitHub App private key specified")
+	}
+	// Generate a JWT using the private key
+	jwt, err := GenerateJWT(clientID, privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Exchange JWT for an installation token
+	token, err := FetchInstallationToken(jwt, installationID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a client using the installation token
+	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tokenClient := oauth2.NewClient(ctx, tokenSource)
+
+	return github.NewClient(tokenClient), nil
+}
+
+// GenerateJWT generates a JWT for a GitHub App.
+func GenerateJWT(clientID string, privateKey string) (string, error) {
+	privkey, err := base64.StdEncoding.DecodeString(privateKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode private key: %v", err)
+	}
+	block, _ := pem.Decode(privkey)
+	if block == nil {
+		return "", fmt.Errorf("failed to decode private key")
+	}
+
+	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse RSA private key: %v", err)
+	}
+
+	now := time.Now().Unix()
+	claims := jwt.MapClaims{
+		"iat": now,       // Issued at time
+		"exp": now + 600, // Expiration time (10 min)
+		"iss": clientID,  // GitHub Client ID
+	}
```

Comment:
Use `jwt.RegisteredClaims` for clarity and avoid doing our own Unix time calculations?